src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -116,9 +119,26 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
 
                 Query highlightQuery = field.fieldOptions().highlightQuery() == null ? context.parsedQuery().query() : field.fieldOptions().highlightQuery();
                 HighlighterContext highlighterContext = new HighlighterContext(fieldName, field, fieldMapper, context, hitContext, highlightQuery);
-                HighlightField highlightField = highlighter.highlight(highlighterContext);
-                if (highlightField != null) {
-                    highlightFields.put(highlightField.name(), highlightField);
+
+                if (highlighter.cannotHighlight(fieldMapper) && fieldNameContainsWildcards) {
+                    // if several fieldnames matched the wildcard then we want to skip those that we cannot highlight
+                    continue;
+                }
+                try {
+                    HighlightField highlightField = highlighter.highlight(highlighterContext);
+                    if (highlightField != null) {
+                        highlightFields.put(highlightField.name(), highlightField);
+                    }
+                } catch (FetchPhaseExecutionException e) {
+                    if (fieldNameContainsWildcards && e.getCause() instanceof BytesRefHash.MaxBytesLengthExceededException) {
```

Comment:
is this a plain highlighter only problem? Not sure, but if so maybe we could handle it in the PlainHighlighter directly rather than in the HighlightPhase?